frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeAddTemplate.tsx

```diff
@@ -140,6 +141,15 @@ const CustomServingRuntimeAddTemplate: React.FC<CustomServingRuntimeAddTemplateP
                 onClick={() => {
                   setIsLoading(true);
                   // TODO: Revert back to pass through api once we migrate admin panel
+                  if (!isServingRuntimeKind(YAML.parse(code))) {
+                    setError(
+                      new Error(
+                        'The "kind" should be "ServingRuntime", and the following fields are required: container, supportedModelFormats.',
+                      ),
+                    );
+                    setIsLoading(false);
+                    return;
+                  }
```

Comment:
Only call `setIsLoading` _after_ this if statement -> right before the async execution begins.
```suggestion
                  if (!isServingRuntimeKind(YAML.parse(code))) {
                    setError(
                      new Error(
                        'The "kind" should be "ServingRuntime", and the following fields are required: container, supportedModelFormats.',
                      ),
                    );
                    return;
                  }
                  setIsLoading(true);
                  // TODO: Revert back to pass through api once we migrate admin panel
```